gcloud-java-resourcemanager/src/test/java/com/google/gcloud/resourcemanager/ProjectTest.java

```diff
@@ -0,0 +1,214 @@
+  private Project project;
+
+  @Before
+  public void setUp() throws Exception {
+    resourceManager = createStrictMock(ResourceManager.class);
+    project = new Project(resourceManager, PROJECT_INFO, POLICY);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    verify(resourceManager);
+  }
+
+  @Test
+  public void testLoad() {
+    expect(resourceManager.get(PROJECT_INFO.id())).andReturn(PROJECT_INFO);
+    expect(resourceManager.getIamPolicy(PROJECT_INFO.id())).andReturn(POLICY);
+    replay(resourceManager);
+    Project loadedProject = Project.load(resourceManager, PROJECT_INFO.id());
+    assertEquals(PROJECT_INFO, loadedProject.info());
+    assertEquals(POLICY, loadedProject.policy());
+  }
+
+  @Test
+  public void testReload() {
+    ProjectInfo newInfo = PROJECT_INFO.toBuilder().addLabel("k3", "v3").build();
+    Policy newPolicy = POLICY.toBuilder().removeBinding(VIEWER_BINDING).build();
+    expect(resourceManager.get(PROJECT_INFO.id())).andReturn(newInfo);
+    expect(resourceManager.getIamPolicy(PROJECT_INFO.id())).andReturn(newPolicy);
+    replay(resourceManager);
+    Project newProject = project.reload();
+    assertSame(resourceManager, newProject.resourceManager());
+    assertEquals(newInfo, newProject.info());
+    assertEquals(newPolicy, newProject.policy());
+  }
+
+  @Test
+  public void testPolicy() {
+    assertEquals(project.policy(), POLICY);
+    replay(resourceManager);
+  }
+
+  @Test
+  public void testInfo() {
+    assertEquals(project.info(), PROJECT_INFO);
+    replay(resourceManager);
+  }
+
+  @Test
+  public void testResourceManager() {
+    assertEquals(project.resourceManager(), resourceManager);
+    replay(resourceManager);
+  }
+
+  @Test
+  public void testDelete() {
+    resourceManager.delete(PROJECT_INFO.id());
+    expectLastCall();
+    replay(resourceManager);
+    project.delete();
+  }
+
+  @Test
+  public void testUndelete() {
+    resourceManager.undelete(PROJECT_INFO.id());
+    expectLastCall();
+    replay(resourceManager);
+    project.undelete();
+  }
+
+  @Test
+  public void testReplace() {
+    ProjectInfo newInfo = PROJECT_INFO.toBuilder().addLabel("k3", "v3").build();
+    expect(resourceManager.replace(newInfo)).andReturn(newInfo);
+    replay(resourceManager);
+    Project newProject = project.replace(newInfo);
+    assertSame(newProject.resourceManager(), resourceManager);
+    assertEquals(newProject.info(), newInfo);
+    assertEquals(newProject.policy(), POLICY);
+  }
+
+  @Test
+  public void testReplaceIamPolicy() {
+    Policy newPolicy = POLICY.toBuilder().removeBinding(VIEWER_BINDING).build();
+    expect(resourceManager.replaceIamPolicy(PROJECT_INFO.id(), newPolicy)).andReturn(newPolicy);
+    replay(resourceManager);
+    Project newProject = project.replaceIamPolicy(newPolicy);
+    assertSame(newProject.resourceManager(), resourceManager);
+    assertEquals(newProject.info(), PROJECT_INFO);
+    assertEquals(newProject.policy(), newPolicy);
```

Comment:
Nit, usually the first parameter is the expected value and the second one the actual.